include/qpid/dispatch/router_core.h

```diff
@@ -23,6 +23,7 @@
 #include <qpid/dispatch/bitmask.h>
 #include <qpid/dispatch/compose.h>
 #include <qpid/dispatch/parse.h>
+typedef struct qdr_core_t         qdr_core_t;
```

Comment:
Please move this back down with the other typedefs.  You can add this line into router.h to solve the compilation issue.